### PR TITLE
rivet: remove deprecated versions and clean up recipe

### PR DESCRIPTION
**Pull request by jmcarcell**
and add a new version

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
I'll be waiting a couple more days to check if anybody has issues with removing versions, but honestly this is a great cleanup. Thanks!

File: var/spack/repos/builtin/packages/rivet/package.py [hunk 1]
```diff
@@ -264,24 +49,20 @@ class Rivet(AutotoolsPackage):
...
+    depends_on("yoda@1.7.0:1.7", when="@3.0.0,3.0.2")
 
     depends_on("hepmc", when="hepmc=2")
     depends_on("hepmc3", when="hepmc=3")
-    depends_on("boost", when="@:2.5.0")
     # TODO: replace this with an explicit list of components of Boost,
     # for instance depends_on('boost +filesystem')
     # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants, when="@:2.5.0")
+    depends_on(Boost.with_default_variants)
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/rivet/package.py]: https://gitlab.com/hepcedar/rivet/-/releases/rivet-2.5.0: there is not dependency on boost anymore.

Comment (wdconinc) [reply var/spack/repos/builtin/packages/rivet/package.py]: "3. Elimination of the Boost library in favour of C++11."

Comment (jmcarcell) [inline var/spack/repos/builtin/packages/rivet/package.py]:
```suggestion
```

Comment (jmcarcell) [reply var/spack/repos/builtin/packages/rivet/package.py]: Deleted!

File: var/spack/repos/builtin/packages/rivet/package.py [hunk 2]
```diff
@@ -264,24 +48,19 @@ class Rivet(AutotoolsPackage):
...
-    depends_on("yoda@1.7.0:1.7", when="@2.6.0,2.7.0,2.7.1,3.0.0,3.0.2")
-    depends_on("yoda@1.5.0:1.5", when="@2.4.1")
+    depends_on("yoda@1.7.0:1.7", when="@3.0.0,3.0.2")
 
     depends_on("hepmc", when="hepmc=2")
     depends_on("hepmc3", when="hepmc=3")
-    depends_on("boost", when="@:2.5.0")
     # TODO: replace this with an explicit list of components of Boost,
     # for instance depends_on('boost +filesystem')
     # See https://github.com/spack/spack/pull/22303 for reference
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/rivet/package.py]: Boost component comment is now not necessary anymore.
```suggestion
```

**Review by wdconinc — APPROVED** [review]
Approving with the one remaining suggested change to the comments only.

**Comment by jmcarcell** [comment]
Comments about boost are now removed. I removed the patch files that are deleted in the recipe and rebased

**Review by wdconinc — APPROVED** [review]
Can confirm this still installs fine without the patches etc.
```
==> Installing rivet-3.1.8-ievss4kztcpepqwt6vxypibrfn5zsctf [120/120]
==> No binary for rivet-3.1.8-ievss4kztcpepqwt6vxypibrfn5zsctf found: installing from source
==> Fetching https://rivet.hepforge.org/downloads/?f=Rivet-3.1.8.tar.bz2
==> No patches needed for rivet
==> rivet: Executing phase: 'autoreconf'
==> rivet: Executing phase: 'configure'
==> rivet: Executing phase: 'build'
==> rivet: Executing phase: 'install'
==> rivet: Successfully installed rivet-3.1.8-ievss4kztcpepqwt6vxypibrfn5zsctf
  Stage: 26.54s.  Autoreconf: 6.15s.  Configure: 13.35s.  Build: 12m 12.96s.  Install: 26.45s.  Post-install: 1.22s.  Total: 13m 27.20s
[+] /opt/software/linux-ubuntu23.04-skylake/gcc-12.3.0/rivet-3.1.8-ievss4kztcpepqwt6vxypibrfn5zsctf
```

**Review by alalazo — APPROVED** [review]

**Pull request merged**